inference_overview.md

```diff
@@ -0,0 +1,449 @@
+
+## Overall flow
+
+### Loading external models
+
+B&A will support loading models inside the TEE. B&A servers will fetch and load
+ad tech models periodically from a connected cloud bucket and make them
+available for inference in the ad tech provided code (e.g. `generateBid()`). The
+loading will be performed in the background, separate from the execution of
+`generateBid()`. After loading, the models will also be pre-warmed in the
+background, so that when the models are used in the request flow, they are
+already warmed.
+
+<figure id = "image-2">
+  <img src = "images/inference_overview_fig2.png"
+  width = "100%"
+  alt = "Loading external models">
+  <figcaption><b>Figure 2.</b> Loading external models</figcaption>
+</figure>
+
+The loaded models will be executed in the supported frameworks (e.g. Tensorflow,
+PyTorch). When `runInference()` is called from within `generateBid()`, the call
+will be forwarded to the corresponding executor to execute the inference
+request, and return the results back to the calling function. These calls occur
+within the confines of a single TEE host, and data never leaves the TEE.
+
+Note that `runInference()will` not call out to the cloud buckets, or trigger
+loading of the model. It will only execute requests against models that have
+already been loaded.
+
+### Flows to support factorized models
+
+To support factorized models, B&A will support data flows that enable
+transferring embeddings produced from the model parts to `generateBid()`, so
+that they can be combined together with the prediction made through the
+`runInference()` call. These flows are based on the existing [request
+processing][2] flows in B&A.
+
+B&A will support these flows as following:
+
+- Support passing embeddings generated as a part of the Contextual RTB auction
+  to `generateBid()` through `buyer_signals`. These embeddings can be produced
+  in real time, or can be looked up from ad tech servers during the contextual
+  RTB request processing.
+- Support model loading and inference from `prepareDataForAdRetrieval()` through
+  the `runInference()` call. `prepareDataForAdRetrieval()` can be used to
+  generate private embeddings that are passed to the [Ad Retrieval service][11]
+  to filter and fetch Ads.
+- Support querying embeddings from the Ad Retrieval service (runs inside TEE)
+  and passing them to `generateBid()`. These embeddings will be produced offline
+  and uploaded to the Ad retrieval service. The keys will define what the
+  embeddings are for, and appropriate keys can be looked up during [retrieval
+  request][10] to fetch the pre-computed embeddings
+- Support model loading and inference in `generateBid()` through the
+  `runInference()` call, as described above. Here, inference runs inside TEE and
+  can use private features based on the inputs of `generateBid()`. This produces
+  an embedding that can be combined with the embeddings from 1) and 2). The data
```

Comment:
what is "1) and 2)" ? Can you make them explicit?

Reply:
done